Add first steps for the ssh troubleshooting (#2400)
CSCfi/csc-user-guide: docs/support/faq/why-cant-i-connect-to-my-vm-in-pouta.md

In order to add a virtual ip, follow the [Post creation step](../../cloud/pouta/launch-vm-from-web-gui.md#post-creation-step) guide.

### port 22: Connection timed out

```sh
$ ssh cloud-user@yy.yy.yyy.yy
ssh: connect to host yy.yy.yyy.yy port 22: Connection timed out
```

If you are not able to connect to your VM, the most common reason for these problems are **Firewalls** and **Security Groups** (Openstack's firewall) that are too restrictive. Newly launched virtual machines will by default block all traffic. You need to create a new security group. The security group must open the SSH port 22 to the ingress traffic.

Please follow the [Firewalls and security](../../cloud/pouta/launch-vm-from-web-gui.md#firewalls-and-security-groups) groups article. If the problem persists you may check the firewall setup of your local institution.

!!! Warning "Permission denied"
    Incorrectly configured Security Groups, can lead to permissions denied errors due to the fact that the VM needs to fetch the public SSH keys on its first start. If the network is not configured properly, the public key may not be added and no access will be configured.

>     FOr this reaseon, you need to make sure that the `default` security group is configured in the VM at creation.

Reviewer: ```suggestion
    For this reason, you need to make sure that the `default` security group is configured in the VM at creation.
```

### Are you using the correct user?

Different distributions are configured with different defaults users. See here the up to date list of [images](/cloud/pouta/images/#images) and their corresponding default users. For example if you are using "Ubuntu 24.04", the correct user is "ubuntu", but if you are using "AlmaLinux-9" the correct one is "almalinux".

It is a common practise for Pouta images, when you try to login as `root`, to get a message back telling you which username to use instead:

```sh
$ ssh root@86.xxx.xxx.xxx
Please login as the user "cloud-user" rather than the user "root".
```

### Which key pair is configured in the Instance? And do you have the matching Private Key?

In order to see which "**Key Pair**" is configured on that VM. Go to <https://pouta.csc.fi/dashboard/project/instances/> and see the key pair name.

![VM Status check](../../cloud/img/instances-keypair.png)

Then go to <https://pouta.csc.fi/dashboard/project/key_pairs> where all **your** keys are configured. The page will give you the list of **public keys**. In the screenshots above, the "**Key Pair**" is `SGC-key`.

![Key pairs listing](../../cloud/img/key-pairs-listing.png)

> Copy the Public key content (`ssh-rsa AAA.... jack@sgc.com`) into a filed called `SGC-key.pub`. You can then calculate the **Public key** fingerprint by:

Reviewer: ```suggestion
Copy the Public key content (`ssh-rsa AAA.... jack@sgc.com`) into a file called `SGC-key.pub`. You can then calculate the **Public key** fingerprint by:
```

```sh
$ ssh-keygen -lf SGC-key.pub
2048 SHA256:FjN0zrymP3mMZzTJ/UJrypmVVcH8Wgok9+JBiBhcvFc no comment (RSA)
```

The example above uses the `SHA256` algorithm by default. You can force it to use `MD5` by doing:

```sh
$ ssh-keygen -lf .ssh/SGC-key.pub -E md5
2048 MD5:eb:b3:eb:ff:65:2f:cb:a5:fa:ab:f4:84:04:a2:d3:9a no comment (RSA)
```

!!! Info "No Key Pair listed"
    It is possible that the VM was created without a "**keypair**` (the corresponding field is empty `-`), in this case you need to [re-create the VM](../../cloud/pouta/launch-vm-from-web-gui.md), this time with a keypair configured.
    ![No Key Pair](../../cloud/img/instance-no-keypair.png)

> Then it is necessary to find the corresponding **Private key**. In Linux and MacOS, private key are normally stored on the `.ssh` folder. You can use the same command as above to generate the fingerprint of the private key:

Reviewer: ```suggestion
Then it is necessary to find the corresponding **Private key**. In Linux and MacOS, private key are normally stored in the `.ssh` folder. You can use the same command as above to generate the fingerprint of the private key:
```

```sh
$ ssh-keygen -lf .ssh/SGC-key
2048 SHA256:FjN0zrymP3mMZzTJ/UJrypmVVcH8Wgok9+JBiBhcvFc no comment (RSA)
```

The fingerprints must match.

!!! Info "Instance log"
    You can also see the finger print of the key from the **Instance log**. Go to [instance page](https://pouta.csc.fi/dashboard/project/instances/) and click in the name of the instance. Then click in the Log tab. You will need to find the lines that begin with `ci-info`. This is the output of the cloud init script.

    ![Instance log](../../cloud/img/log-instance.png)

>     You will be able to see the username, the file where the keys are configured and the list of finger prints of the configured keys. In the exampe above the output shgows the `MD5` algorithm.

Reviewer: ```suggestion
    You will be able to see the username, the file where the keys are configured and the list of finger prints of the configured keys. In the example above the output shows the `MD5` algorithm.
```
